unified_planning/io/pddl_writer.py

```diff
@@ -180,7 +196,7 @@ def _write_domain(self, out: IO[str]):
                 i = 0
                 for p in f.signature():
                     if p.is_user_type():
-                        params.append(f' ?p{str(i)} - {p.name()}') # type: ignore
+                        params.append(f' ?p{str(i)} - {p.name() if p.name() != "object" else self.object_freshname}') # type: ignore
```

Comment:
you are using this piece of code everywhere, maybe incapsulate it in a utility function?